webrender/src/frame_builder.rs

```diff
@@ -1535,9 +1535,10 @@ impl<'a> LayerRectCalculationAndCullingPass<'a> {
             };
 
             if bounding_rect.is_none() {
-                bounding_rect =
-                    Some(clip_info.xf_rect.as_ref().map_or_else(DeviceIntRect::zero,
-                                                                |x| x.bounding_rect))
+                bounding_rect = Some(match clip_info.xf_rect {
```

Comment:
Right. My concern is - if we only patch with `screen_rect` here, then there might be other places where `xf_rect` going out of bounds will become a problem. So I figured fixing it upstream would be safer than patching downstream.